fatiando/gravmag/transform.py

```diff
@@ -503,6 +507,105 @@ def derivz(x, y, data, shape, order=1, method='fft'):
     return deriv[padx: padx + nx, pady: pady + ny].ravel()
 
 
+def power_density_spectra(x, y, data, shape):
+    r"""
+    Calculates the Power Density Spectra of a 2D gridded potential field
+    through the FFT:
+
+    .. math::
+
+        \Phi_{\Delta T}(k_x, k_y) = | F\left{\Delta T \right}(k_x, k_y) |^2
+
+    .. note:: Requires gridded data.
+
+    .. note:: x, y, z and height should be in meters.
+
+    Parameters:
+
+    * x, y : 1D-arrays
+        The x and y coordinates of the grid points
+    * data : 1D-array
+        The potential field at the grid points
+    * shape : tuple = (nx, ny)
+        The shape of the grid
+
+    Returns:
+
+    * kx, ky : 2D-arrays
+        The wavenumbers of each Power Density Spectra point
+    * pds : 2D-array
+        The Power Density Spectra of the data
+    """
+    kx, ky = _fftfreqs(x, y, shape, shape)
+    pds = abs(numpy.fft.fft2(numpy.reshape(data, shape)))**2
+    return kx, ky, pds
+
+
+def radial_average(kx, ky, pds, max_radius=None, ring_width=None):
+    r"""
+    Calculates the average of the Power Density Spectra points that falls
+    inside concentric rings around the zero wavenumber. By default, the
+    biggest ring is the bigger one to have at least one point of both axes,
+    and the rings' width is set as the maximum of :math:`\Delta k_x` and
+    :math:`\Delta k_y`, being them the equidistances of the kx and ky arrays.
+
+    The width of the rings and the inner radius of the biggest ring can be
+    changed by setting the optional parameters ring_width and max_radius,
+    respectively.
+
+    .. note:: It's recommended to use the outputs of the
+              power_density_spectra function as input of this one.
+
+    Parameters:
+
+    * kx, ky : 2D-arrays
+        The x and y coordinates of the grid points
+    * data : 1D-array
+        The potential field at the grid points
+    * shape : tuple = (nx, ny)
+        The shape of the grid
+    * max_radius : float (optional)
+        Changes the default inner radius of the biggest ring.
```

Comment:
It's better to explain the parameters in the order 1) say what this is (Inner radius of the biggest ring) 2) what default is (The largest value of x.max() and y.max()) then 3) what changing this does to the output and recommend a good value.